src/compiler/diagnosticMessages.json

```diff
@@ -2751,7 +2751,12 @@
     "Resolving real path for '{0}', result '{1}'": {
         "category": "Message",
         "code": 6130
-    },    
+    },
+    "Cannot compile modules using option '{0}' unless the '--module' flag is provided with a valid module type.": {

```

Comment:
Be more explicit - we know exactly which module types are valid, so let's go with

`...unless the '--module' flag is 'amd' or 'system'.`

and maybe UMD if that's supported too - I always forget.